src/indexer/handleEvent.ts

```diff
@@ -550,97 +557,70 @@ async function handleEvent(
           }
         );
 
-        // Insert or update unique application contributor
-        const applicationContributors = db.collection<Contributor>(
-          `rounds/${event.args.roundAddress}/applications/${applicationId}/contributors`
-        );
+      await db
+        .collection<Application>(
+          `rounds/${event.args.roundAddress}/applications`
+        )
+        .updateById(applicationId, (project) => ({
+          ...project,
+          amountUSD: project.amountUSD + amountUSD,
+          votes: project.votes + 1,
+          uniqueContributors:
+            project.uniqueContributors + (isNewapplicationContributor ? 1 : 0),
+        }));
 
-        const isNewapplicationContributor =
-          await applicationContributors.upsertById(
-            event.args.voter,
-            (contributor) => {
-              if (contributor) {
-                return {
-                  ...contributor,
-                  amountUSD: contributor.amountUSD + amountUSD,
-                  votes: contributor.votes + 1,
-                };
-              } else {
-                return {
-                  id: event.args.voter,
-                  amountUSD,
-                  votes: 1,
-                };
-              }
-            }
-          );
+      await db
+        .collection<Vote>(
+          `rounds/${event.args.roundAddress}/applications/${applicationId}/votes`
+        )
+        .insert(vote);
 
-        await db
-          .collection<Application>(
-            `rounds/${event.args.roundAddress}/applications`
+      const contributorPartitionedPath = vote.voter
+        .split(/(.{6})/)
+        .filter((s: string) => s.length > 0)
+        .join("/");
+
+      await Promise.all([
```

Comment:
Since `Promise.all` runs promises in parallel, given the nature of these operations, is there a chance that a different order of resolution might result in different db state? That shouldn't be the case with `updateById`, I'm mostly wondering about `insert`.